mantle/cmd/kola/kola.go

```diff
@@ -576,26 +584,57 @@ func syncFindParentImageOptions() error {
 }
 
 // Note this is a no-op if the decompressed dest already exists.
-func downloadImageAndDecompress(url, compressedDest string) (string, error) {
+func downloadImageAndDecompress(url, compressedDest string, skipSignature bool) (string, error) {
 	var decompressedDest string
-	if strings.HasSuffix(compressedDest, ".xz") {
+	if strings.HasSuffix(compressedDest, ".xz") || strings.HasSuffix(compressedDest, ".gz") {
 		// if the decompressed file is already present locally, assume it's
 		// good and verified already
-		decompressedDest = strings.TrimSuffix(compressedDest, ".xz")
+		decompressedDest = strings.TrimSuffix(strings.TrimSuffix(compressedDest, ".xz"), ".gz")
 		if exists, err := util.PathExists(decompressedDest); err != nil {
 			return "", err
 		} else if exists {
 			return decompressedDest, nil
 		} else {
-			if err := sdk.DownloadCompressedSignedFile(decompressedDest, url, nil, "", util.XzDecompressStream); err != nil {
-				return "", err
+			if !skipSignature {
+				if err := sdk.DownloadCompressedSignedFile(decompressedDest, url, nil, "", util.XzDecompressStream); err != nil {
+					return "", err
+				}
+			} else {
+				req, err := http.NewRequest("GET", url, nil)
+				if err != nil {
+					return "", err
+				}
+
+				resp, err := http.DefaultClient.Do(req)
+				if err != nil {
+					return "", err
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					return "", fmt.Errorf("url %s status: %s", url, resp.Status)
+				}
+				gzr, err := gzip.NewReader(resp.Body)
+				if err != nil {
+					return "", err
+				}
+				dst, err := os.OpenFile(decompressedDest, os.O_WRONLY|os.O_CREATE, 0666)
+				if err != nil {
+					return "", err
+				}
+				defer dst.Close()
+				prefix := filepath.Base(decompressedDest)
+				if _, err := util.CopyProgress(capnslog.INFO, prefix, dst, gzr, resp.ContentLength); err != nil {
+					return "", err
+				}
```

Comment:
Can we put this in the sdk as e.g. `DownloadCompressedFile`?

Reply:
I looked at this and yes it would be a good cleanup but I think we want to rip all this out and just fork off `coreos-installer download` or so, right?  Can I just do that as a followup?

Reply:
Hmm, will `coreos-installer download` eventually learn to download RHCOS artifacts too?